data_steward/cdr_cleaner/cleaning_rules/unit_normalization.py

```diff
@@ -0,0 +1,302 @@
+    /* when transform_value_as_number is null due to left join */
+  ELSE
+  value_as_number
+END
+  AS value_as_number,
+  value_as_concept_id,
+  COALESCE(set_unit_concept_id,
+    unit_concept_id) AS unit_concept_id,
+  CASE transform_value_as_number
+    WHEN "(1/x)" THEN 1/range_low
+    WHEN "(x-32)*(5/9)" THEN (range_low-32)*(5/9)
+    WHEN "*0.02835" THEN range_low * 0.02835
+    WHEN "*0.394" THEN range_low * 0.394
+    WHEN "*0.4536" THEN range_low * 0.4536
+    WHEN "*1" THEN range_low * 1
+    WHEN "*10" THEN range_low * 10
+    WHEN "*10^(-1)" THEN range_low * 0.1
+    WHEN "*10^(-2)" THEN range_low * 0.01
+    WHEN "*10^(3)" THEN range_low * 1000
+    WHEN "*10^(-3)" THEN range_low * 0.001
+    WHEN "*10^(6)" THEN range_low * 1000000
+    WHEN "*10^(-6)" THEN range_low * 0.000001
+    /* when transform_value_as_number is null due to left join */
+  ELSE
+  range_low
+END
+  AS range_low,
+  CASE transform_value_as_number
+    WHEN "(1/x)" THEN 1/range_high
+    WHEN "(x-32)*(5/9)" THEN (range_high-32)*(5/9)
+    WHEN "*0.02835" THEN range_high * 0.02835
+    WHEN "*0.394" THEN range_high * 0.394
+    WHEN "*0.4536" THEN range_high * 0.4536
+    WHEN "*1" THEN range_high * 1
+    WHEN "*10" THEN range_high * 10
+    WHEN "*10^(-1)" THEN range_high * 0.1
+    WHEN "*10^(-2)" THEN range_high * 0.01
+    WHEN "*10^(3)" THEN range_high * 1000
+    WHEN "*10^(-3)" THEN range_high * 0.001
+    WHEN "*10^(6)" THEN range_high * 1000000
+    WHEN "*10^(-6)" THEN range_high * 0.000001
+    /* when transform_value_as_number is null due to left join */
+  ELSE
+  range_high
+END
+  AS range_high,
+  provider_id,
+  visit_occurrence_id,
+  measurement_source_value,
+  measurement_source_concept_id,
+  unit_source_value,
+  value_source_value
+FROM
+  `{{project_id}}.{{dataset_id}}.{{measurement_table}}`
+LEFT JOIN
+  `{{project_id}}.{{dataset_id}}.{{unit_table_name}}`
+USING
+  (measurement_concept_id,
+    unit_concept_id)""")
+
+
+class UnitNormalization(BaseCleaningRule):
+    """
+    Units for labs/measurements will be normalized..
+    """
+
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = 'Units for labs/measurements will be normalized using unit_mapping.csv.'
+        super().__init__(issue_numbers=['DC414', 'DC700'],
+                         description=desc,
+                         affected_datasets=[cdr_consts.DEID_CLEAN],
+                         affected_tables=['measurement'],
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id)
+
+    def get_mapping_list(self, unit_mappings_list):
```

Comment:
Since this is a draft PR, I don't know if you're finished with this or not.  But this isn't being used anywhere right now.

Reply:
my bad this will be removed.